### PR TITLE
Specify terminfo dirs

### PR DESCRIPTION
**Pull request by keuv-grvl**
Inspired by https://github.com/conda/conda-recipes/pull/223/files and https://lists.gnu.org/archive/html/bug-ncurses/2011-06/msg00020.html

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -21,6 +21,8 @@ do
 	    --enable-termcap \
 	    --enable-pc-files \
 	    --with-termlib \
+	    --with-terminfo-dirs="/etc/terminfo:/lib/terminfo:/usr/share/terminfo:$PREFIX/share/terminfo" \
+	    --with-default-terminfo-dir="/usr/share/terminfo" \
```
Comment (jakirkham) [inline recipe/build.sh]: Seems like the default should be ours, no?

Comment (jakirkham) [inline recipe/build.sh]: Done.

**Comment by jakirkham** [comment]
Please bump the build number as well.

**Comment by jakirkham** [comment]
Please add the license file using [`license_file`]( http://conda.pydata.org/docs/building/meta-yaml.html#license-file ).

**Comment by jakirkham** [comment]
@mingwandroid, could you please take a look at this and share your thoughts?

**Comment by mingwandroid** [comment]
If you build `--with-termlib` (which I do not since it caused a lot of pain for me) then you probably want the ones in $PREFIX first. If you don't build `--with-termlib` then you probably want to have the ones in $PREFIX last.

Perhaps both building `termlib` *and* setting the terminfo-dirs appropriately would fix my issues but I'm not really convinced. `ncurses` is fairly complicated, relying on and also setting various env. vars.

My take was to defer to the system as much as I could for everything above `ncurses` itself, so my preference was to remove `--with-termlib` and not pass `--with-terminfo-dirs` at all.

The only useful advice I can give is to test on a multitude of different Linux installations and if it works on them all, then I'd call that good enough, but my instinct would be for you guys to use my recipe as reference here:

https://github.com/mingwandroid/conda-recipes/tree/master/ncurses

.. because if it isn't broke (and is well tested) then don't *fix* it.

**Comment by jakirkham** [comment]
However we can't use your recipe as dropping `--with-termlib` will break tons of things at this point. Plus this breaks with the general trend of using that option on Linux distros. It would be great if we could figure out why things on your end won't work with that option honestly.

File: recipe/build.sh [hunk 2]
```diff
@@ -21,6 +21,8 @@ do
 	    --enable-termcap \
 	    --enable-pc-files \
 	    --with-termlib \
+	    --with-terminfo-dirs="/etc/terminfo:/lib/terminfo:/usr/share/terminfo:$PREFIX/share/terminfo" \
```
Comment (jakirkham) [inline recipe/build.sh]: So the suggestion is to move `$PREFIX/share/terminfo` to the front of the list, which seems reasonable.

**Comment by jakirkham** [comment]
No we had fixed that in PR ( https://github.com/conda-forge/ncurses-feedstock/pull/19 ). AFAICT the previous build and this build of `ncurses` work fine in the `continuumio/miniconda` image.

Please see issue ( https://github.com/ContinuumIO/anaconda-issues/issues/455 ) for a bit more history on why we have our own termlib.

**Comment by jakirkham** [comment]
> I find it hard to follow these multiple threads of discussion but what I did find seemed to circle back to what you already said "We did it because that's what Linux distros do".

While there is some of that. This not the crux of the problem.

The problem is the system may supply a newer copy of the terminfo from a newer `ncurses` than we are using (e.g. 6 vs. 5). So we do need to supply our own terminfo db and we definitely want to use ours first (if not only our).

If we must, we can try to fallback to the system as proposed here. Though it may causes issues if the system has a newer version.

**Comment by mingwandroid** [comment]
I build on CentOS 5.11 then test on that, then I test on a fully up to date ArchLinux system (which uses ncurses-6). This procedure gives me a good level of confidence about the compatibility of my packages.

It seems to me, from https://github.com/conda-forge/ncurses-feedstock/pull/19, that an incorrect partial fix was applied (after I had shared my experience that removing `--with-termlib` worked well) and now another attempt is being made to do something in the same vein.

However the important thing is that if your way proves itself to work well, and you run with it, then your `ncurses` package will break everything in defaults that is linked to our `ncurses`, so as soon as someone installs `htop`, it becomes a support issue for me with defaults' Python and R language.

Whatever we do for ncurses should be done in common so that we are not incompatible, and in this case, I got there first and tried to advise.

**Comment by jakirkham** [comment]
We have been using `--with-termlib` since this package was added back in April. We also discussed dropping it in PR ( https://github.com/conda-forge/ncurses-feedstock/pull/10 ) before packaging `python` and other things that relied on it. The decision was not to do so. We worked with other user groups downstream (like bioconda) to add `.pc` files to make it easier to work with in PR ( https://github.com/conda-forge/ncurses-feedstock/pull/11 ).

As everyone is reliant on `ncurses` here being built using `--with-termlib`, switching will causes problems not just at conda-forge, but for other communities reliant on us and this way of doing things. I fail to see how switching it now is a good plan that avoids breakage for the larger conda community.

The proposal here seems like a simple reasonable change that should help out in a few other cases by searching some system paths. Feel free to test other things against this change. That being said, unless there are some clear blockers, I'm in favor of going forward with this strategy.

File: recipe/build.sh (hunk 2 above)
Comment (keuv-grvl) [inline recipe/build.sh]: Done

**Comment by jakirkham** [comment]
Going to put a hold on this while we work on `htop` some more.

**Comment by keuv-grvl** [comment]
I admit it has been a bit technical for me here :)

Anyway, we worked on my [PR](https://github.com/conda-forge/staged-recipes/pull/1598) to add `htop` to the conda-forge. Now it builds successfully (locally and within a Docker container) without changing the `ncurses` recipe, which is nice.

So I would close this PR.
Thank you for your help!
